Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Registration.hs

```diff
@@ -436,12 +438,14 @@ getRegistrationTokenE :: EsqDBFlow m r => Id SR.RegistrationToken -> m SR.Regist
 getRegistrationTokenE tokenId =
   RegistrationToken.findById tokenId >>= fromMaybeM (TokenNotFound $ getId tokenId)
 
-createPerson :: (EncFlow m r, EsqDBFlow m r, DB.EsqDBReplicaFlow m r, Redis.HedisFlow m r, CacheFlow m r) => AuthReq -> Text -> Maybe Text -> Maybe Version -> Maybe Version -> Id DMerchant.Merchant -> m SP.Person
+createPerson :: (EncFlow m r, EsqDBFlow m r, DB.EsqDBReplicaFlow m r, Redis.HedisFlow m r, CacheFlow m r, CoreMetrics m) => AuthReq -> Text -> Maybe Text -> Maybe Version -> Maybe Version -> Id DMerchant.Merchant -> m SP.Person
 createPerson req mobileNumber notificationToken mbBundleVersion mbClientVersion merchantId = do
   person <- buildPerson req mobileNumber notificationToken mbBundleVersion mbClientVersion merchantId
+  createPersonStats <- makePersonStats person
   -- DB.runTransaction $ do
   _ <- Person.create person
   _ <- QDFS.create $ makeIdlePersonFlowStatus person
+  _ <- QPS.create createPersonStats
```

Comment:
Is it necessary to create it here? You have to always check `(all = 0)` before filling data when you could just check if it exists or not. When stats API is called for a new person, we could show all 0. This could be done to save some data for users who just register and don't use but fine, you may choose to ignore it also depending on what you like more.

Reply:
Sure